src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/IncentivesService.kt

```diff
@@ -67,4 +104,25 @@ class IncentivesService(
       auditModule = incentiveEntity.auditModuleName,
     )
   }
+
+  private fun mapIncentiveModel(dto: CreateIncentiveRequest, offenderBooking: OffenderBooking): Incentive {
+
+    val sequence = offenderBooking.getNextSequence()
+
+    val iepLevel = iepLevelRepository.findById(IEPLevel.pk(dto.iepLevel))
+      .orElseThrow(BadDataException("Invalid IEP type from incentives: ${dto.iepLevel}"))
+
+    val location = agencyLocationRepository.findById(dto.agencyId)
+      .orElseThrow(BadDataException("Prison with id=${dto.agencyId} does not exist"))
+
```

Comment:
I think you also need to check the incentive level exist for the prison as well (sorry can't remember table)
In fact, if you just check IEP level/location exists then the 2 above calls are redundant I think?